exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java

```diff
@@ -136,6 +136,55 @@ NativeHandle intoPatch() {
     }
   }
 
+  /**
+   * Creates in-memory checkpoint that can be used to rollback changes.
+   */
+  void createCheckpoint() {
+    checkState(nativeCanRollback(getNativeHandle()),
+        "This fork does not support checkpoints");
+
+    // TODO: Invalidate all indexes created with the fork or the Core won't let us
+    //  do anything.
+
+    nativeCreateCheckpoint(getNativeHandle());
+  }
+
+  /**
+   * Rollbacks changes to the latest checkpoint. Affects only changes made with
+   * this particular Fork instance.
+   *
+   * <p>If no checkpoints was created, rollbacks all changes made by this fork.
+   */
+  void rollback() {
+    checkState(nativeCanRollback(getNativeHandle()),
+        "This fork does not support rollbacks");
+
+    // TODO: Invalidate all indexes created with the fork or the Core won't let us
+    //  do anything.
+
+    nativeRollback(getNativeHandle());
+  }
+
+  /**
+   * Creates in-memory checkpoint that can be used to rollback changes.
+   */
+  private static native void nativeCreateCheckpoint(long nativeHandle);
+
+  /**
+   * Rollback changes to the latest checkpoint. Affects only changes made with
+   * this particular Fork instance.
+   */
+  private static native void nativeRollback(long nativeHandle);
+
+  /**
+   * Returns true if creating checkpoints and performing rollbacks is
+   * possible with this particular Fork instance.
+   *
+   * @see #createCheckpoint()
+   * @see #rollback()
+   */
+  private static native boolean nativeCanRollback(long nativeHandle);
```

Comment:
If all native methods are placed in the end of the file, I'd keep them in the order they are used (i.e., put the new ones after the *patch).

Reply:
Done